src/controllers/graphql/resolvers/post/postResolver.js

```diff
@@ -1,28 +1,110 @@
+    getAllPosts: [Post]
+    getPostById(id: ID!): Post
+    getFollowerPosts: [Post]
+    getHundredxPosts: [Post]
+    # getTextPosts: [Post]
+    # getNewsPosts: [Post]
+    # getVideoPosts: [Post]
+    # getTrendingPosts(trending: String): [Post]
+    # getPostsByTags(tags: [String]): [Post]
+  }
+
+  type Mutation {
+    createPost(post: Post!): Post
+    updatePost(post: Post!): Post
+    deletePost(id: ID!): boolean # Return boolean if delete was successful or not
+  }
+
+  type Post {
+    id: ID!
+    title: String
+    body: String
+    authorId: String
+    # Other post values
+  }
+`;
 
-module.exports = {
+const resolvers = {
   Query: {
     getAllPosts: async (_, args) => {
-      return await Post.findAll({order: [['hot', 'ASC']], limit: FETCH_TOP_X_POSTS});
+      return await Post.findAll({
+        order: [['hot', 'ASC']],
+        limit: FETCH_TOP_X_POSTS
+      });
     },
-    getPost: async (_,args) => {
+
+    getPost: async (_, args) => {
       return await Post.findById(args.postId);
     },
+
     getFollowerPosts: async (_, args, { authenticatedUserInstance }) => {
-      return !authenticatedUserInstance
-        ? new AuthenticationError('ERROR_GETTING_FOLLOWER_POSTS')
-        : await authenticatedUserInstance.getMyFollowersPosts();
+      if (!authenticatedUserInstance) {
+        throw new AuthenticationError('INVALID_USER');
+      }
+      return await authenticatedUserInstance.getMyFollowersPosts();
     },
+
     getHundredxPosts: async (_, args) => {
       return await PostService.fetchHundredxResteemedPosts();
     }
+
+    // getTextPosts: async(_, args) => {
+    //   return await PostService.fetchTextPosts();
+    // },
+
+    // getNewsPosts: async(_, args) => {
+    //   return await PostService.fetchNewsPosts();
+    // },
+
+    // getVideoPosts: async(_, args) => {
+    //   return await PostService.fetchVideoPosts();
+    // },
+
+    // getTrendingPosts: async(_, { trending }) => {
+    //   return await PostService.getTrendingPosts(trending);
+    // },
+
+    // getPostsByTags: async(_, { tags }) => {
+    //   return await PostService.getPostsByTags(tags);
+    // }
   },
+  Mutation: {
+    // createPost: async (_, { post }, { authenticatedUserInstance }) => {
+    //   if (!authenticatedUserInstance) {
+    //     throw new AuthenticationError('INVALID_USER');
+    //   }
+    //   return PostService.createPost(post);
+    // },
+
+    // updatePost: async (_, { post }, { authenticatedUserInstance }) => {
+    //   if (!authenticatedUserInstance) {
+    //     throw new AuthenticationError('INVALID_USER');
+    //   }
+    //   return PostService.updatePost(post);
+    // },
+
+    // deletePostById: async(_, { id }, { authenticatedUserInstance  }) => {
+    //   if (!authenticatedUserInstance) {
+    //     throw new AuthenticationError('INVALID_USER');
+    //   }
+    //   return PostService.deletePostById(id);
+    // },
+  },
+  /** Same with this. What is this doing?  */
```

Comment:
Wasn't sure what this Post object is doing

Reply:
Very confused.  Where is this code coming from.  I think you wrote that code?

Reply:
https://github.com/merivercap/100x-Backend/commit/b3a283dd33d68c12913ebe8608c143666a7a6b92

negative

Reply:
This is my fault though because this means I didn't code review thoroughly. I should have asked this question when you made the PR

Reply:
Ah ok.  No its cool dude.  Sorry, I was just confused what you were asking about.  The User hash for example allows us to do something like

```
getUserById {
  posts {
     ... ///bla bla bla post info 
  } 
}


  ```User: {
    posts(author) {
      return author.getPosts();
    }
  }```